test/deploy.py

```diff
@@ -0,0 +1,569 @@
+    ) -> None:
+        """Send Files (recursively) to VM"""
+        if not self.client:
+            raise AssertionError("LXD Client not available to runner.")
+
+        if not self.client.instances.exists(vm_name):
+            raise AssertionError("VM {} does not exist.".format(vm_name))
+
+        cmd = [
+            "lxc",
+            "file",
+            "push",
+            src_path,
+            "{}{}".format(vm_name, target_path),
+            "-r",
+        ]
+        try:
+            subprocess.check_call(cmd)
+        except subprocess.CalledProcessError as e:
+            print("Failed Pushing {} to {}: Output {}"
+                  .format(src_path, vm_name, e))
+            raise e
+
+    def create_storage_volume(
+        self,
+        size="10GB",
+        count=3,
+        pool_name=deploy_tag,
+    ) -> list:
+        """Create Storage Volume from test storage pool"""
+        if not self.client:
+            raise AssertionError("LXD Client not available to runner.")
+
+        if not self.client.storage_pools.exists(pool_name):
+            raise AssertionError("Storage Pool {} does not exist."
+                                 .format(pool_name))
+
+        storage_pool = self.client.storage_pools.get(pool_name)
+        volumes = []
+        for iter in range(0, count):
+            vol_name = "vol-" + _get_random_string(4)
+            config = {
+                "name": vol_name, "type": "custom", "content_type": "block"
+            }
+
+            # Creating Storage Volume.
+            print("Creating Storage Volume {}".format(vol_name))
+            storage_pool.volumes.create(config, wait=True)
+            # Save volume name for returning
+            volumes.append(vol_name)
+
+        self.model["volumes"] = volumes
+        return volumes
+
+    def exec_remote_script(
+        self, vm_name: string, relative_script_path: string,
+        params=[], op_print=True
+    ) -> None:
+        """Execute a remote script on LXD VM."""
+        if not self.client:
+            raise AssertionError("LXD Client not available to runner.")
+
+        if not self.client.instances.exists(vm_name):
+            raise AssertionError("VM {} does not exist.".format(vm_name))
+        cmd = [
+            "bash", self.complete_repo_path + "/" + relative_script_path,
+            *params
+        ]
+        if op_print:
+            print("Executing on {}: CMD: {}".format(vm_name, cmd))
+        self.check_call_on_vm(vm_name, cmd)
+
+    def install_apt_package(
+        self,
+        vm_name: string,
+        relative_script_path="test/scripts/deploy_helper.sh",
+    ) -> None:
+        """Installs the required packages on lxd vm."""
+        self.exec_remote_script(vm_name, relative_script_path, ["install_apt"])
+
+    def grow_root_partition(self, vm_name: string) -> None:
+        """Use Growpart utility to increase root partition size."""
+        self.check_call_on_vm(vm_name, ["growpart", "/dev/sda", "2"])
```

Comment:
How safe is it to assume the device names (and its partitions) ?

Reply:
Root device is the first volume so it always attaches as /dev/sda. First partitions is for /boot/efi, so i believe it is safe to assume it will always be /dev/sda2.
```
root@oci-testing-160Z-3PTL:~# lsblk
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
sda      8:0    0 18.6G  0 disk 
├─sda1   8:1    0  100M  0 part /boot/efi
└─sda2   8:2    0 18.5G  0 part /
```